### ISSUE TITLE
memoize db keygen is inconsistent

### ISSUE THREAD

**Issue by cevans87** [cevans87/atools#42]
When memoizing to db, we use the sha256 hash of the string of all input arguments to the function. Some arguments may be equivalent but may stringify differently due to factors such as:

- hash seed changing between program executions, changing the order that items print in sets/dicts
- dictionary items being specified in different orders, like `{'a': 'b', 'c': 'd'}` vs `{'c': 'd', 'a': 'b'}`

**Comment by cevans87** [comment]
I've tried to tackle this in a these ways, and this is why it doesn't seem to work.

- Find a way to use the builtin `hash()` without using the PYTHONHASHSEED
-- This requires digging into the C implementation of hash, and might just not be doable. There be dragons here, and I think I'm rightfully scared.
- Require a `__memo_hash__` function that returns a consistent string representation or hash value that is not affected by PYTHONHASHSEED
-- I have no idea how to enforce that the value is not affected by PYTHONHASHSEED. I also couldn't figure out how to do this in a way that wasn't really awkward for the user.

I've decided that the best route here is to add documentation with a stern warning and clear instructions about how to avoid pitfalls when using `keygen=...` with `db=True`. I hope to find a perfect solution in the future.
